amp.php

```diff
@@ -157,3 +160,20 @@ function _amp_bootstrap_customizer() {
 	}
 }
 add_action( 'plugins_loaded', '_amp_bootstrap_customizer', 9 );
+
+/**
+ * Redirects the old AMP URL to the new AMP URL.
+ * If post slug is updated the amp page with old post slug will be redirected to the updated url.
+ *
+ * @param  string $link New URL of the post.
+ *
+ * @return string $link URL to be redirected.
+ */
+function amp_redirect_to_new_url( $link ) {
+
+	if ( is_amp_endpoint() ) {
+		$link = trailingslashit( $link ) . AMP_QUERY_VAR;
```

Comment:
One possible issue here is if someone is using a different permalink structure for AMP URLs, this will break for them.

I don't see a simple fix for this though because the filter does not pass the post id (so we may just have to accept it as a known issue): https://core.trac.wordpress.org/browser/tags/4.7/src/wp-includes/query.php#L904